product_pricelist_supplierinfo/tests/test_product_supplierinfo.py

```diff
@@ -133,3 +135,37 @@ def test_pricelist_based_price_round(self):
         self.assertAlmostEqual(
             self.pricelist.get_product_price(self.product, 1, False), 20.0,
         )
+
+    def test_supplierinfo_per_variant(self):
+        tmpl = self.env['product.template'].create({
+            'name': 'Test Product'
+        })
+        attr = self.env.ref('product.product_attribute_1')
+        attr_val1 = self.env.ref('product.product_attribute_value_1')
+        attr_val2 = self.env.ref('product.product_attribute_value_2')
+        self.env['product.template.attribute.line'].create({
+            'product_tmpl_id': tmpl.id,
+            'attribute_id': attr.id,
+            'value_ids': [(6, 0, [attr_val1.id, attr_val2.id])],
+        })
+        tmpl.create_variant_ids()
```

Comment:
Why not use?
<pre>
tmpl = self.env['product.template'].create({
    'name': 'Test Product',
    'attribute_line_ids': [[0, 0, {
        'attribute_id': self.env.ref('product.product_attribute_1').id,
        'value_ids': [
            (4, self.env.ref('product.product_attribute_value_1').id),
            (4, self.env.ref('product.product_attribute_value_2').id),
        ],
    ],
})
</pre>

Reply:
Can't find any reason why not. Yours is indeed more compact. I will change it. Thanks.